lib/igniter/code/common.ex

```diff
@@ -226,20 +230,29 @@ defmodule Igniter.Code.Common do
   \"\"\"
   ```
   """
-  @spec add_code(Zipper.t(), String.t() | Macro.t(), :after | :before) :: Zipper.t()
-  def add_code(zipper, new_code, placement \\ :after)
+  @spec add_code(Zipper.t(), String.t() | Macro.t(), add_code_opts() | atom()) :: Zipper.t()
+  def add_code(zipper, new_code, opts \\ [placement: :after, expand_env?: true])
+
+  def add_code(_zipper, _new_code, placement) when is_atom(placement) do
+    raise ArgumentError, """
+    The `placement` argument is deprecated. Use an opts list of `placement: #{inspect(placement)}` instead.
+    """
```

Comment:
Let's deprecate this instead of raising, so this should change to something like:

```elixir
def add_code(zipper, new_code, placement) when is_atom(placement) do
  Logger.warning("""
  Passing an atom as the third argument to `Igniter.Code.Common.add_code/3` is deprecated in favor of an options list.
  
  Instead of:
  
      Igniter.Code.Common.add_code(zipper, new_code, #{inspect(placement)})
  
  You should now write:
  
      Igniter.Code.Common.add_code(zipper, new_code, placement: #{inspect(placement)})
  """)

  add_code(zipper, new_code, placement: placement)
end
```

Reply:
I like this much more thank you! Fixed in [`eec48a9` (#151)](https://github.com/ash-project/igniter/pull/151/commits/eec48a91e45d90c49a67e02621c1bf23d5cf5f32)